test/DatasetAuthorization.js

```diff
@@ -633,4 +645,922 @@ describe("0300: DatasetAuthorization: Test access to dataset", () => {
+      ownerGroup: "group1",
+    };
+
+    return request(appUrl)
+      .post("/api/v3/Datasets")
+      .send(newDataset)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${accessTokenAdmin}` })
+      .expect("Content-Type", /json/)
+      .then((res) => {
+        res.statusCode.should.not.be.equal(200);
+        res.body.should.not.have.property("pid");
+      });
+  });
+
+  it("0530: admin can add a new raw dataset with specified pid and different owner group", async () => {
+    const datasetWithPid = {
+      ...TestData.RawCorrect,
+      pid: uuidv4(),
+      ownerGroup: "group1",
+    };
+
+    return request(appUrl)
+      .post("/api/v3/Datasets")
+      .send(datasetWithPid)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${accessTokenAdmin}` })
+      .expect(200)
+      .expect("Content-Type", /json/)
+      .then((res) => {
+        res.body.should.have.property("pid").and.equal(datasetWithPid.pid);
+        res.body.should.have.property("owner").and.be.string;
+        res.body.should.have.property("type").and.equal("raw");
+      });
+  });
+
+  it("0540: admin cannot add a new raw dataset with specified invalid pid and different owner group", async () => {
+    const datasetWithPid = {
+      ...TestData.RawCorrect,
+      pid: "this-is-invalid-pid-1",
+      ownerGroup: "group1",
+    };
+
+    return request(appUrl)
+      .post("/api/v3/Datasets")
+      .send(datasetWithPid)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${accessTokenAdmin}` })
+      .expect("Content-Type", /json/)
+      .then((res) => {
+        res.statusCode.should.not.be.equal(200);
+        res.body.should.not.have.property("pid");
+      });
+  });
+
+  it("0550: admin cannot add a new invalid raw dataset with specified pid and different owner group", async () => {
+    const invalidDatasetWithPid = {
+      ...TestData.RawWrong_1,
+      pid: uuidv4(),
+      ownerGroup: "group1",
+    };
+
+    return request(appUrl)
+      .post("/api/v3/Datasets")
+      .send(invalidDatasetWithPid)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${accessTokenAdmin}` })
+      .expect("Content-Type", /json/)
+      .then((res) => {
+        res.statusCode.should.not.be.equal(200);
+        res.body.should.not.have.property("pid");
+      });
+  });
+
+  it("0560: admin cannot add a new invalid raw dataset with specified invalid pid and different owner group", async () => {
+    const invalidDatasetWithInvalidPid = {
+      ...TestData.RawWrong_1,
+      pid: "this-is-invalid-pid-1",
+      ownerGroup: "group1",
+    };
+
+    return request(appUrl)
+      .post("/api/v3/Datasets")
+      .send(invalidDatasetWithInvalidPid)
+      .set("Accept", "application/json")
+      .set({ Authorization: `Bearer ${accessTokenAdmin}` })
+      .expect("Content-Type", /json/)
+      .then((res) => {
+        res.statusCode.should.not.be.equal(200);
```

Comment:
For invalid dataset creation, I'd suggest to use equal to 400 status, to make the test more precise.